src/sources/nginx_metrics/mod.rs

```diff
@@ -298,6 +310,20 @@ mod integration_tests {
                 user: "vector".to_owned(),
                 password: "vector".to_owned(),
             }),
+            ProxyConfig::default(),
+        )
+        .await
+    }
+
+    #[tokio::test]
+    async fn test_stub_status_with_proxy() {
+        test_nginx(
+            "http://vector_nginx:8000/basic_status",
+            None,
+            ProxyConfig {
+                http: Some("http://localhost:3128".into()),
+                ..Default::default()
+            },
```

Comment:
I might just add a comment here that this is the test we have that verifies simple proxy support in Vector. Users might otherwise think this test exists for all components. Maybe something like:

```
// This integration test verifies that proxy support is wired up correctly in Vector
// It is the only test of its kind
```